src/test/java/uk/gov/companieshouse/api/accounts/transformer/StocksTransformerTest.java

```diff
@@ -0,0 +1,224 @@
+package uk.gov.companieshouse.api.accounts.transformer;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.stocks.CurrentPeriodEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.stocks.PreviousPeriodEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.stocks.StocksDataEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.stocks.StocksEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.stocks.CurrentPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.stocks.PreviousPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.stocks.Stocks;
+
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class StocksTransformerTest {
+
+    private static final Long PAYMENTS_ON_ACCOUNT_CURRENT_PERIOD = 1L;
+    private static final Long STOCKS_CURRENT_PERIOD = 2L;
+    private static final Long TOTAL_CURRENT_PERIOD = 3L;
+
+    private static final Long PAYMENTS_ON_ACCOUNT_PREVIOUS_PERIOD = 10L;
+    private static final Long STOCKS_PREVIOUS_PERIOD = 20L;
+    private static final Long TOTAL_PREVIOUS_PERIOD = 30L;
+
+    private static final String ETAG = "etag";
+    private static final String KIND = "kind";
+
+    private StocksTransformer stocksTransformer = new StocksTransformer();
+
+    @Test
+    @DisplayName("Tests transformer with empty rest object returns null values ")
+    public void testTransformerWithEmptyRestObject() {
+
+        StocksEntity stocksEntity = stocksTransformer
+                .transform(new Stocks());
+
+        assertNotNull(stocksEntity);
+        assertNull(stocksEntity.getData().getEtag());
+        assertEquals(new HashMap<>(), stocksEntity.getData().getLinks());
+    }
+
+    @Test
+    @DisplayName("Tests transformer with empty previous period Rest Object")
+    public void testRestToEntityTransformerWithEmptyPreviousPeriodRestObject() {
+
+        Stocks stocks = new Stocks();
+
+        stocks.setEtag(ETAG);
+        stocks.setKind(KIND);
+        stocks.setLinks(new HashMap<>());
+
+        CurrentPeriod currentPeriod = createCurrentPeriodRestObject();
+
+        stocks.setCurrentPeriod(currentPeriod);
```

Comment:
I would do the same here as you do in the other test `stocks.setCurrentPeriod(createCurrentPeriodRestObject());`